apps/worker/src/request_processor/request.processor.service.ts

```diff
@@ -0,0 +1,78 @@
+import { InjectRedis } from '@liaoliaots/nestjs-redis';
+import { Processor, WorkerHost, OnWorkerEvent } from '@nestjs/bullmq';
+import { Injectable, Logger } from '@nestjs/common';
+import { DelayedError, Job } from 'bullmq';
+import Redis from 'ioredis';
+import { ConfigService } from '../../../api/src/config/config.service';
+import { AnnouncementTypeDto, BroadcastDto, IRequestJob, ProfileDto, QueueConstants, ReplyDto, UpdateDto } from '../../../../libs/common/src';
+import { IpfsService } from '../../../../libs/common/src/utils/ipfs.client';
+
+@Injectable()
+@Processor(QueueConstants.REQUEST_QUEUE_NAME)
+export class RequestProcessorService extends WorkerHost {
+  private logger: Logger;
+
+  constructor(
+    @InjectRedis() private redis: Redis,
+    private configService: ConfigService,
+    private ipfsService: IpfsService,
+  ) {
+    super();
+    this.logger = new Logger(this.constructor.name);
+  }
+
+  async process(job: Job<IRequestJob, any, string>): Promise<any> {
+    this.logger.log(`Processing job ${job.id} of type ${job.name}`);
+    this.logger.debug(job.asJSON());
+    const assets = this.getAssetReferencesFromRequestJob(job.data);
+
+    const pinnedAssets = assets.map((cid) => this.ipfsService.getPinned(cid));
+    const pinnedResult = await Promise.all(pinnedAssets);
+    this.logger.debug(pinnedResult);
+    // if any of assets does not exists delay the job for a future attempt
+    if (pinnedResult.some((buffer) => !buffer)) {
+      await this.delayJobAndIncrementAttempts(job);
+    } else {
+      // TODO: create attachments from assets
+    }
```

Comment:
Assuming we have some TODOs here to also queue the requests to batchQueue?

Reply:
that is the #11 ticket

Reply:
I see , great